test/runtests.jl

```diff
@@ -1,8 +1,29 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
-using Test, StyledStrings
+using Test
 
-import StyledStrings: SimpleColor, Face, styled
+using StyledStrings: StyledStrings, Legacy, SimpleColor, FACES, Face,
+    @styled_str, styled, StyledMarkup, getface, addface!, loadface!, resetfaces!
+using .StyledMarkup: MalformedStylingMacro
+using Base: AnnotatedString, AnnotatedChar, AnnotatedIOBuffer, annotations
+
+# For output testing
+
+const kitty = Base.TermInfo(read(joinpath(@__DIR__, "terminfos", "kitty"), Base.TermInfoRaw))
+const vt100 = Base.TermInfo(read(joinpath(@__DIR__, "terminfos", "vt100"), Base.TermInfoRaw))
+
+function with_terminfo(fn::Function, tinfo::Base.TermInfo)
```

Comment:
Is the terminfo license MIT compatible? Would want to be certain before distributing this as part of the stdlib.

Reply:
Hmm, good point.

From what I read, it looks like the vt100 file will follow the general ncurses database license, but the kitty terminfo file is originally sourced from https://github.com/kovidgoyal/kitty which is GPL-3. There's also the aspect where it look like it may be a stretch whether compiled terminfo files themselves are copyrightable, since they are a machine-readable description of the method of operation for the terminal, with little to no room for variation in expression.

In perhaps a similar vein, I think it's worth noting that recipes are not copyrightable, with a comment from a [US government FAQ](https://www.copyright.gov/circs/circ33.pdf) on this reading

> A recipe is a statement of the ingredients and procedure required for making a dish of food. A mere listing of ingredients or contents, or a simple set of directions, is uncopyrightable. As a result, the Office cannot register recipes consisting of a set of ingredients and a process for preparing a dish.

Given a terminfo file is a statement of recognised ANSI codes and their forms, I wouldn't be surprised if somebody more legally knowledgeable than me were to say the logic applied to a list of ingredients in a recipe also applies to a list of Termcap names and their ANSI codes.

Reply:
I agree with you in principle, that it appears difficult to describe these binaries as a work that falls under copyright, but distributing something which originates in any form from a GPL-3 licensed codebase spooks me. 

I'm not a lawyer and I can't say that your interpretation is wrong. Still, without advice from an actual lawyer, it may be prudent to avoid the risk entirely by removing the files from the source. The fact that its copyright status is not immediately apparent is enough to warrant extreme caution, in my opinion.

Is it possible to generate the terminfos files as a part of the testing script? Or to fetch them from the original distributors? In this way, the files can still be used without the risk of distributing something that may or may not be licensed under GPL-3.

Reply:
I've just switched to the wezterm terminfo, which should have all the capabilities we test for but comes from a MIT-licenced codebase.

Reply:
Ah darn it, it fails to set the `setrgbf` parameter.

Reply:
Screw it, I'll make a custom terminfo.